File: backend/HRM.Api/Models/LeaveRequest.cs
using HRM.Api.Constants;

namespace HRM.Api.Models
{
    public class LeaveRequest
    {
        public int LeaveRequestID { get; set; }
        public int EmployeeID { get; set; }
        public int ManagerID { get; set; }
        public int LeaveTypeID { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool IsHalfDayStart { get; set; }
        public bool IsHalfDayEnd { get; set; }
        public decimal TotalDays { get; set; }
        public string? Reason { get; set; }
        public string Status { get; set; } = LeaveStatus.Pending;
        public DateTime RequestedDate { get; set; } = DateTime.Now;
        public string? AttachmentPath { get; set; }

        // Navigation
        public virtual Employee? Employee { get; set; }
        public virtual Employee? Manager { get; set; }
        public virtual LeaveType? LeaveType { get; set; }
        public virtual ICollection<LeaveRequestHistory> Histories { get; set; }
    }
}

File: backend-java/hrm-utility/src/main/java/com/hrm/utility/entity/LeaveRequest.java
package com.hrm.utility.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "LeaveRequests")
@Data // Lombok để tự tạo getter/setter
public class LeaveRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "LeaveRequestID")
    private Integer leaveRequestID;

    @Column(name = "ManagerID")
    private Integer managerID; // Đây là ID của người quản lý cần duyệt đơn này

    @Column(name = "StartDate")
    private LocalDate startDate;

    @Column(name = "EndDate")
    private LocalDate endDate;

    @Column(name = "TotalDays")
    private BigDecimal totalDays;

    @Column(name = "Status")
    private String status;

    @Column(name = "RequestedDate")
    private LocalDateTime requestedDate;

    @Column(name = "ApprovalNote")
    private String approvalNote;

    @Column(name = "ApprovedDate")
    private LocalDateTime approvedDate;

    // Join để lấy thông tin người xin nghỉ (Employee)
    @ManyToOne
    @JoinColumn(name = "EmployeeID", insertable = false, updatable = false)
    private Employee employee;

    // Join để lấy tên loại nghỉ (LeaveType)
    @ManyToOne
    @JoinColumn(name = "LeaveTypeID", insertable = false, updatable = false)
    private LeaveType leaveType;
}